packages/mds-agency/request-handlers.ts

```diff
@@ -135,7 +135,7 @@ export const getVehicleById = async (req: AgencyApiRequest, res: AgencyApiRespon
 
   const { cached } = req.query
 
-  const { provider_id } = res.locals
+  const { provider_id } = res.locals.scopes.includes('vehicles:read') ? req.query : res.locals
```

Comment:
`GitHub` won't let me comment, but I think lines `142-148` need to change as well. Currently, if the request is scoped with `vehicles:read` then the request will return `404` _unless_ it also contains a matching _provider_id_ in the query string which seems unnecessary. The solution could be to adjust the `if` condition or perhaps pass an _optional_ provider_id to `readPayload`.

Reply:
SGTM, fixed